libraries/GCS_MAVLink/GCS_Common.cpp

```diff
@@ -1142,7 +1151,7 @@ uint16_t GCS_MAVLINK::get_reschedule_interval_ms(const deferred_message_bucket_t
         // we are sending requests for waypoints, penalize streams:
         interval_ms *= 4;
     }
-    if (AP_HAL::millis() - ftp.last_send_ms < 500) {
+    if (AP_HAL::millis() - ftp.last_send_ms < 1000) {
```

Comment:
It seems odd that we would use 1 second here but 3 seconds to time out a FTP session.

Reply:
I've only looked at QGroundControl which has a 1 second ack/nak timeout with 3 retries.  Does Mission Planner use a 3 second timeout?

Reply:
I was thinking of this timeout:
https://github.com/ArduPilot/ardupilot/blob/01b0e0c27cf91169fb00c5e18197e613539a5339/libraries/GCS_MAVLink/GCS_FTP.cpp#L35

Reply:
(our responses crossed here)  OK.  I think things should work OK with this changed to 3 seconds.  The stream rates will just take 3 seconds to go back to normal instead of 1.  Should we move the define to a header file and use FTP_SESSION_TIMEOUT instead of 500 or 1000 in GCS_Common.cpp?

Reply:
Actually, after more consideration, I think there is little reason that this timeout should be the same as the session timeout.  This is essentially a heuristic and should be based on the expected behavior of the GCS, including radio latency, but doesn't need to match the session timeout which really needs to be conservative (long) as discarding the open state too early would be bad.   With the current GCS behavior, I think 1 second is fine.  
On the other hand, it would be easier for slower radios to make this work well if the GCS FTP retry timeout was a bit longer, say 2 seconds.  Bumping this from 500 to 1000 was enough for mLRS which was already working hard to keep buffered data to a minimum.  But, ELRS is even slower than mLRS.  Should we increase this to 2 seconds in anticipation that some work may also be required on the GCS side to help out ELRS?

Reply:
My thinking is that if FTP thinks the session is still active then we should still lower its data rates. However a quick test shows the MAVProxy terminates the session after param download whereas QGC and MP do not. If we get the GCS to terminate correctly there should be no problem in using the longer time out here.